### PR TITLE
Support undisturbed ground temperature models on GroundHeatExchangerVertical

### LINKED ISSUE #4930: Support undisturbed ground temperature models on GroundHeatExchangerVertical
Enhancement Request
--------------
Do for `GroundHeatExchanger:Vertical` what was done for `GroundHeatExchanger:HorizontalTrench` in https://github.com/NREL/OpenStudio/pull/4717: add "Undisturbed Ground Temperature Model" field for which either the `Site:GroundTemperature:Undisturbed:KusudaAchenbach` or `Site:GroundTemperature:Undisturbed:Xing` object could be referenced.

### PR DESCRIPTION
**Pull request by joseph-robertson**
Pull request overview
---------------------

Fixes #4930.

Checklist:
- [x] Add "Undisturbed Ground Temperature Model" to idd
- [x] Update model files
- [x] Update model tests
- [x] Update FT
- [x] Update FT tests
- [x] Update VT
- [x] Update VT tests

### Pull Request Author

 - [x] Model API Changes / Additions
 - [x] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [x] Model API methods are tested (in `src/model/test`)
 - [x] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [x] If a new object or method, added a test in NREL/OpenStudio-resources: https://github.com/NREL/OpenStudio-resources/pull/199
 - [x] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [ ] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [ ] All new and existing tests passes
 - [ ] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [x] If change to an IDD file, add the label `IDDChange`
 - [ ] If breaking existing API, add the label `APIChange`
 - [x] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

File: src/energyplus/ForwardTranslator/ForwardTranslateGroundHeatExchangerVertical.cpp [hunk 1]
```diff
@@ -166,28 +176,12 @@ namespace energyplus {
...
-
-    if ((value = modelObject.groundThermalConductivity())) {
-      groundIdfObject.setDouble(Site_GroundTemperature_Undisturbed_KusudaAchenbachFields::SoilThermalConductivity, value.get());
-    }
-
-    groundIdfObject.setDouble(Site_GroundTemperature_Undisturbed_KusudaAchenbachFields::SoilDensity, 920.0);
+    // UndisturbedGroundTemperatureModelName
+    idfObject.setString(GroundHeatExchanger_SystemFields::UndisturbedGroundTemperatureModelName, s.get());
 
-    if ((value = modelObject.groundThermalHeatCapacity())) {
```
Comment (kbenne) [inline src/energyplus/ForwardTranslator/ForwardTranslateGroundHeatExchangerVertical.cpp:180]: What happens to these fields if they are set on the GroundHeatExchangerVertical system? If looks like they might be ignored. Maybe the implementation of these methods like GroundHeatExchangerVertical::groundThermalHeatCapacity() should map to the contained ground temperature model?

File: src/energyplus/Test/GroundHeatExchangerVertical_GTest.cpp [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  Model m;
+
+  GroundHeatExchangerVertical ghx(m);
+
+  EXPECT_TRUE(ghx.setDesignFlowRate(0.004));
+  EXPECT_TRUE(ghx.setNumberofBoreHoles(100));
+  EXPECT_TRUE(ghx.setBoreHoleLength(80.0));
+  EXPECT_TRUE(ghx.setBoreHoleRadius(0.7E-01));
+  EXPECT_TRUE(ghx.setGroundThermalConductivity(0.7));
+  EXPECT_TRUE(ghx.setGroundThermalHeatCapacity(0.3E+07));
```
Comment (kbenne) [inline src/energyplus/Test/GroundHeatExchangerVertical_GTest.cpp:48]: Like I mentioned. Do these methods do anything now?

File: src/model/GroundHeatExchangerVertical.hpp [hunk 3]
```diff
@@ -93,6 +95,8 @@ namespace model {
 
     bool isGFunctionReferenceRatioDefaulted() const;
 
+    ModelObject undisturbedGroundTemperatureModel() const;
```
Comment (kbenne) [inline src/model/GroundHeatExchangerVertical.hpp:98]: Similar question as above on the ctor, why a generic ModelObject? The name `undisturbedGroundTemperatureModel()` implies I'm going to get a thing of that type.

File: src/energyplus/ForwardTranslator/ForwardTranslateGroundHeatExchangerVertical.cpp (hunk 1 above)
Comment (joseph-robertson) [reply src/energyplus/ForwardTranslator/ForwardTranslateGroundHeatExchangerVertical.cpp:180]: `ghxVert.groundThermalConductivity()` and `ghxVert.groundThermalHeatCapacity()` both still get translated above.

`ghxVert.groundTemperature()` previously was only translated to the `groundTempModel`. `AverageSoilSurfaceTemperature` for the kusuda object is now handled by its own `averageSoilSurfaceTemperature()` method.

This is basically the same approach we used for the `HorizontalTrench` implementation - the `ghxVert` gets the `soilThermalConductivity()` and `soilSpecificHeat`, and then kusuda FTs its own `averageSoilSurfaceTemperature()`.

File: src/energyplus/Test/GroundHeatExchangerVertical_GTest.cpp (hunk 2 above)
Comment (joseph-robertson) [reply src/energyplus/Test/GroundHeatExchangerVertical_GTest.cpp:48]: These 2 still translate to the ghxVert object (see explanation above).

File: src/model/GroundHeatExchangerVertical.hpp [hunk 4]
```diff
@@ -39,6 +39,8 @@ namespace model {
 
     explicit GroundHeatExchangerVertical(const Model& model);
 
+    explicit GroundHeatExchangerVertical(const Model& model, const ModelObject& undisturbedGroundTemperatureModel);
```
[the remark this answers is not in the document]
Comment (joseph-robertson) [reply src/model/GroundHeatExchangerVertical.hpp:42]: I think we want to receive a ModelObject here. There are others, for example the `Site:GroundTemperature:Undisturbed:FiniteDifference`, that we may want to wrap in the future.

Here's a related comment on this: https://github.com/NREL/OpenStudio/pull/4717#discussion_r1107164769

File: src/model/GroundHeatExchangerVertical.hpp (hunk 3 above)
Comment (joseph-robertson) [reply src/model/GroundHeatExchangerVertical.hpp:98]: See explanation above.

File: src/energyplus/ForwardTranslator/ForwardTranslateGroundHeatExchangerVertical.cpp (hunk 1 above)
Comment (kbenne) [reply src/energyplus/ForwardTranslator/ForwardTranslateGroundHeatExchangerVertical.cpp:180]: sorry missed the above. Got it.

File: src/model/GroundHeatExchangerVertical.hpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (kbenne) [reply src/model/GroundHeatExchangerVertical.hpp:42]: ok. I'd definiately favor being consistent with the horizontal ground heat exchanger, and I see the logic. Sounds good to me.

File: src/model/GroundHeatExchangerVertical.hpp (hunk 3 above)
Comment (kbenne) [reply src/model/GroundHeatExchangerVertical.hpp:98]: got it.

**Review by kbenne — APPROVED** [review]

File: developer/doc/ReleaseNotes/OpenStudio_Release_Notes_3_7_0_TDB.md [hunk 5]
```diff
@@ -78,6 +78,9 @@ You can also refer to the [OpenStudio SDK Python Binding Version Compatibility M
     * Fix #4695 - AirLoopHVACUnitarySystem: Supply Air Flow Rate Method During <XXX> Operation should be set via related setters/autosize
     * Breaks the return of `supplyAirFlowRateMethodDuringCoolingOperation` and `supplyAirFlowRateMethodDuringHeatingOperation`: now returns `std::string` instead of `boost::optional<std::string>`
     * Deprecates many set/reset methods
+* [#4932](https://github.com/NREL/OpenStudio/pull/4932) - Support undisturbed ground temperature models on GroundHeatExchangerVertical
+    * Fix #4930 - Support undisturbed ground temperature models on GroundHeatExchangerVertical
+    * Update `GroundHeatExchanger:Vertical` to actually use the Ground Temeprature Model field
```
Comment (joseph-robertson) [inline developer/doc/ReleaseNotes/OpenStudio_Release_Notes_3_7_0_TDB.md:83]: Reminder to fix spelling of "Temeprature". (This was copied from the 3.6.0 release notes where it is misspelled.)

**Pull request merged**